opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -442,17 +442,30 @@ namespace Opm
         upper.resize( A.N() );
         inv.resize( A.N() );
 
-        lower.reserveAdditional( 2*A.N() );
+        // Count the lower and upper matrix entries.
+        size_type numLower = 0;
+        size_type numUpper = 0;
+        const auto endi = A.end();
+        for (auto i=A.begin(); i!=endi; ++i) {
+          const size_type iIndex = i.index();
+          for (auto j = (*i).begin(); j.index() < iIndex; ++j) {
```

Comment:
Here (*i).size should be used as before, e.g. numLower += (*i).size();

Reply:
Would not that be the size of the entire row, i.e. both lower and upper entries?

Reply:
You are right. I think we should just use A.nonzeroes() to estimate the number of entries of lower and upper.